subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/SmokeContinuousIntegrationTest.groovy

```diff
@@ -170,19 +156,14 @@ class SmokeContinuousIntegrationTest extends AbstractContinuousIntegrationTest {
     def "does not detect changes in filtered file tree inputs when first matching file is created"() {
         def sources = file("sources").createDir()
         sources.file("sub/some/excluded").createFile()
-        def includedFile = sources.file("sub/some/included.txt")
 
         buildFile << taskOnlyIncludingTxtFiles
 
         when:
         succeeds("myTask")
         then:
         outputContains("includedFiles: 0")
-
-        when:
-        includedFile.text = "created"
-        then:
-        noBuildTriggered()
+        exitsContinuousBuildSinceNotWatchingAnyLocations()
```

Comment:
These tests all exit early now, no? So they are not really testing different behaviors, or do they?

Reply:
Like we discussed, they are still testing different behaviours. Only `noBuildTriggered()` has been replaced by `exitsContinuousBuildSinceNotWatchingAnyLocations()`, that means we exit instead of waiting for a change which we can't receive.